tests/components/blebox/test_init.py

```diff
@@ -0,0 +1,60 @@
+"""BleBox devices setup tests."""
+
+import logging
+
+import blebox_uniapi
+
+from homeassistant.components.blebox.const import DOMAIN
+from homeassistant.config_entries import ENTRY_STATE_NOT_LOADED, ENTRY_STATE_SETUP_RETRY
+
+from .conftest import mock_config, patch_product_identify
+
+
+async def test_setup_failure(hass, caplog):
+    """Test that setup failure is handled and logged."""
+
+    patch_product_identify(None, side_effect=blebox_uniapi.error.ClientError)
+
+    entry = mock_config()
+    entry.add_to_hass(hass)
+
+    caplog.set_level(logging.ERROR)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert "Identify failed at 172.100.123.4:80 ()" in caplog.text
+    assert entry.state == ENTRY_STATE_SETUP_RETRY
+
+
+async def test_setup_failure_on_connection(hass, caplog):
+    """Test that setup failure is handled and logged."""
+
+    patch_product_identify(None, side_effect=blebox_uniapi.error.ConnectionError)
+
+    entry = mock_config()
+    entry.add_to_hass(hass)
+
+    caplog.set_level(logging.ERROR)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert "Identify failed at 172.100.123.4:80 ()" in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)